src/Mvc/Mvc.RazorPages/src/Builder/RazorPagesEndpointRouteBuilderExtensions.cs

```diff
@@ -75,7 +76,9 @@ public static IEndpointConventionBuilder MapFallbackToPage(this IEndpointRouteBu
             EnsureRazorPagesServices(endpoints);
 
             // Called for side-effect to make sure that the data source is registered.
-            GetOrCreateDataSource(endpoints).CreateInertEndpoints = true;
+            var pageDataSource = GetOrCreateDataSource(endpoints);
+            pageDataSource.CreateInertEndpoints = true;
+            RegisterInCache(endpoints.ServiceProvider, pageDataSource);
```

Comment:
nit: Should setting CreateInertEndpoints/RegisterInCache be moved into GetOrCreateDataSource? Or another helper method created?

That would reduce duplication.

Reply:
It can't, CreateInertEndpoints and RegisterInCache need to be called only when mapping dynamic endpoints.